src/pointer.cr

```diff
@@ -129,6 +129,16 @@ struct Pointer(T)
     (self + offset).value = value
   end
 
+  {% if flag?(:x86_64) %}
+    private def to_size_t(size : Int) : UInt64
+        size.to_u64
+    end
+  {% else %}
+    private def to_size_t(size : Int) : UInt32
+        size.to_u32
+    end
+  {% end %}
+
```

Comment:
Yes, we need to introduce generic SizeT and OffsetT types, or change `Intrinsics` contracts, but it should be discussed and introduced by another PR.

This PR just fixes contract implementation.